File: algorithms/python/1409_Queries_on_a_Permutation_With_Key.py
class Solution:
    def processQueries(self, queries: List[int], m: int) -> List[int]:
        perm = list(range(m,0,-1))
        retList = []
        
        for query in queries:
            retList.append(m-1 - perm.index(query))
            perm.pop(m-1 - retList[-1])
            perm.append(query)
            
        return retList

File: algorithms/java/1409_Queries_on_a_Permutation_With_Key.java
class Solution {
    public int[] processQueries(int[] queries, int m) {
        List<Integer> permu = new ArrayList<>();
        for(int i= 1 ; i <= m ; ++i)
            permu.add(i);
        
        int[] retList = new int[queries.length];
        int idx = 0;
        
        for(int i = 0 ; i < queries.length ; ++i) {
            Integer query = queries[i];
            retList[idx++] = permu.indexOf(query);
            permu.remove(query);
            permu.add(0, query);
        }
        
        return retList;
    }
}
